archlinux/plasma5-applets-kde-githubissues.install

```diff
@@ -0,0 +1,23 @@
+post_install() {
+    for i in $(cat /etc/passwd|grep -v nologin|grep -v halt|grep -v shutdown|awk -F":" '{ print $1}'); do
```

Comment:
First time seeing something like this. Got spooked when I first saw `/etc/passwd`. Is this a standard way of iterating users?!

You don't need to iterate all users anyways. All you need to do is unzip to `/usr/share/plasma/plasmoids/org.kde.plasma.githubissues` and it will be accessible to all users.

You appear to be... installing to `~/.local/share/plasma/plasmoids` for every user, when it's already located in `/usr/share/plasma/plasmoids` which is a bit silly.

Reply:
No, this PKGBUILD installs this plasmoid to /usr/share/plasma/plasmoids/. 
I've tried to directly unzip it to /usr/share/plasma/plasmoids/, but it only made the plasmoid appear in the plasmoid list, unable to be dragged onto the desktop. "kpackagetool5" can only install the plasmoid for the current user, but all *.install scripts are executed as root. So I have to iterate every user and install this plasmoid for them.

Reply:
You need to use

```
sudo kpackagetool5 --global --type Plasma/Applet --install package
```

```
$ kpackagetool5 --help
...
  -g, --global                        For install or remove, operates on
                                      packages installed for all users.
```

This should probably work for a QML only widget like githubissues. Other widgets that require a C++ Qt plugin to work use a `CMakeList.txt`.

-----

eventcalendar just unzips to `/usr/share/...` but if that works properly for new users. https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=plasma5-applets-eventcalendar

volumewin7mixer patches in a `CMakeLists.txt` which I should probably be in the repo in the first place. Using the CMakeLists.txt pattern requires the `extra-cmake-modules` dependency though.

* https://aur.archlinux.org/cgit/aur.git/tree/CMakeLists.txt?h=plasma5-applets-volumewin7mixer
* https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=plasma5-applets-volumewin7mixer

It's similar to the redshift-control widget.

* https://github.com/kotelnik/plasma-applet-redshift-control/blob/master/CMakeLists.txt
* https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=plasma5-applets-redshift-control-git

Reply:
Thanks. I'll update the PKGBUILD.

Reply:
Also, the `PKGBUILD` will need to run the `translate/build` script to convert the plaintext `*.po` to the binary `.mo` files under `contents/locale/...`. Otherwise the translations for the widget won't work.

Reply:
I haven't considered locale in the PKGBUILD yet... I may add it later.

Reply:
Does `sudo kpackagetool5 --global --type Plasma/Applet --install package` work?

-----

I assumed `plasma_install_package(package org.kde.plasma.volumewin7mixer)` in the `CMakeLists.txt` pattern worked. That's weird.

> I've tried to directly unzip it to /usr/share/plasma/plasmoids/, but it only made the plasmoid appear in the plasmoid list, unable to be dragged onto the desktop.

That sounds... weird. I'm wondering if it's "cached" the `~/.local/share/plasma/plasmoids` as the location of the widget, and is failing to load it from `/usr/share/`... that does not sound right though...

Have you restarted `plasmashell` since you've fussed with installing? Uninstall the widget, then run `killall plasmashell; kstart5 plasmashell` (or relog).

Reply:
I found the exact problem. I messed up the install dir. It should be `com.github.zren.githubissues` not `ord.kde.plasma.githubissues`. 

There is no need for a *.install. Sorry for the waste of time.

Reply:
Aaaah, I totally missed that the namespace was wrong too. Oops.